### PR TITLE
[SDP-1077] Implement DistributionAccountResolver and use it throughout the code

### PR DESCRIPTION
**Pull request by marcelosalloum**
### What

Implement DistributionAccountResolver and use it throughout the code.

I tested using `DISTRIBUTION_SIGNER_TYPE=DISTRIBUTION_DB` and it works as expected! Here's what I validated in my tests:
- [x] One different distribution account address was created per new tenant
- [x] Once founded, it was possible to create trustlines to that dist account using the UI dashboard (per tenant)
- [x] The TSS executed a payments using transactions associated with that tenant and the source account was the one that belonged to the correct tenant
- [x] I did an e2e test and got the payment synched back to the SDP and updated in the AnchorPlatform, after a fix in the AP PATCH request

### Why

This is the last piece needed to have one distribution account per tenant, and to use `DISTRIBUTION_SIGNER_TYPE-DISTRIBUTION_DB`.

### Pending

- [x] Add `DISTRIBUTION_PUBLIC_KEY` to the k8s deployments

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [ ] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [x] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/transactionsubmission/engine/signing/signature_service.go [hunk 1]
```diff
@@ -75,6 +76,10 @@ func NewSignatureService(opts SignatureServiceOptions) (SignatureService, error)
 		return SignatureService{}, fmt.Errorf("invalid distribution signer type %q", distSignerType)
 	}
 
+	if opts.DistributionAccountResolver == nil {
+		return SignatureService{}, fmt.Errorf("distribution account resolver cannot be nil")
+	}
+
```
Comment (marcelosalloum) [inline internal/transactionsubmission/engine/signing/signature_service.go:82]: Making the resolver mandatory here triggered most of the test changes.

File: internal/dependencyinjection/distribution_accoiunt_resolver_test.go
Comment (marwen-abid) [inline internal/dependencyinjection/distribution_accoiunt_resolver_test.go:1]: Typo in name of the file `distribution_accoiunt_resolver_test.go`

File: internal/dependencyinjection/distribution_accoiunt_resolver.go
Comment (marwen-abid) [inline internal/dependencyinjection/distribution_accoiunt_resolver.go:1]: typo in the name of the file. `distribution_accoiunt_resolver.go`

File: internal/anchorplatform/platform_api_service.go [hunk 2]
```diff
@@ -135,8 +137,14 @@ func (a *AnchorPlatformAPIService) updateAnchorTransactions(ctx context.Context,
 		return fmt.Errorf("error making request to anchor platform: %w", err)
 	}
 
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		log.Ctx(ctx).Errorf("reading response body: %v", err)
+	}
+	defer response.Body.Close()
```
Comment (marwen-abid) [inline internal/anchorplatform/platform_api_service.go:144]: 🫡 Thank you for fixing this!

File: cmd/serve.go [hunk 3]
```diff
@@ -496,6 +503,14 @@ func (c *ServeCommand) Command(serverService ServerServiceInterface, monitorServ
 			}
 			serveOpts.AnchorPlatformAPIService = apAPIService
 
+			// Setup Distribution Account Resolver
+			distAccResolverOpts.AdminDBConnectionPool = adminDBConnectionPool
+			distAccResolver, err := signing.NewDistributionAccountResolver(distAccResolverOpts)
```
Comment (marwen-abid) [inline cmd/serve.go:508]: ❓Any reason why we're not using DI here ? 
```golang
distAccResolver, err := di.NewDistributionAccountResolver(ctx, distAccResolverOpts)
```

**Review by marwen-abid — APPROVED** [review]
LGTM! Thank you for all these changes

File: cmd/serve.go (hunk 3 above)
Comment (marcelosalloum) [reply cmd/serve.go]: Thanks for catching that. FIxed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/212/commits/06a131833b1a60dedf6a9dde092cfe1f3bf79a86.

File: internal/dependencyinjection/distribution_accoiunt_resolver.go
Comment (marcelosalloum) [reply internal/dependencyinjection/distribution_accoiunt_resolver.go:1]: Fixed in https://github.com/stellar/stellar-disbursement-platform-backend/pull/212/commits/dfb0779cf99d7cd45c3a9943ea6a8e89bf244fa4

**Pull request merged**